backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+***
+Test#4 - 261 - 309
+get data from userAnswer
+***
+
+***
+pre-req: 
+    !isSuperUser($userid) && isset($_SESSION["submission-$courseid-$coursevers-$duggaid-$moment"]) == TRUE &&
+    isset($_SESSION["submission-password-$courseid-$coursevers-$duggaid-$moment"] == TRUE
+***
+
+$pre-values = {
+        $hash = “AcIj-Ld6”
+        $hashpwd = “OjOEh8R”
+        $variant = 13
+    }
+
+    //Get the selected user by hash
+    SELECT password,timesSubmitted,timesAccessed,grade from userAnswer WHERE hash=$hash;
+
+***
+send parameters input to showDuggaservice.php
+***
+send 
+    {
+        $opt = “SAVDU”
+        $hash = “AcIj-Ld6”
+    }
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #5 - 279 - 287
+update submitted dugga
+
+***
+Pre-req: 
+    !isset($grade) || !$grade > 1 &&
+    isset($dbpwd) && strcmp($hashpwd,$dbpwd) === 0
+***
+
+***
+Pre-req variables
+***
+$pre-values =
+    {
+        $aid = 999
+        $cid = 2
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $moment = 2002
+        $answer = NULL
+    }
+
+    SQL-query: INSERT INTO userAnswer(aid,cid,hash,password,moment,useranswer) VALUES (pre-values[0], pre-values[1], pre-values[2], pre-values[3], pre-values[4], pre-values[5]);
+
+* Get newly created data
+    SELECT * WHERE hash=pre-values[2];
+
+***
+* Update information on userAnswer
+* send parameters input to showDuggaservice.php
+***
+send
+    {
+        $opt = SAVDU
+        $hash = “dfg4zxc5”
+        $haspwd = “asdfasdf”
+        $answer = “im updated”
+    }
+
+    SQL-query: "UPDATE userAnswer SET submitted=NOW(), useranswer=$answer, timesSubmitted=timesSubmitted+1 WHERE hash=$hash AND password=$hashpwd
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM userAnswer WHERE hash = pre-values[2];
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #6 - 293 - 307
+submit dugga
+
+***
+Pre-req: !isset($grade) || !$grade > 1
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $opt = “SAVDU”
+        $cid = 2
+        $coursevers = 52432
+        $duggaid = 1
+        $moment = 2002
```

Comment:
what happens if moment 2002 no longer exists ?
you should make an insert into table listentries for this test as well. 

dont know if you can use this or not, bit this is an entire that i used in one of my tests 

```
$preValuesListentries = {   1885, 
                            "Inserttobedeleted",
                            "UNK",
                            4,
                            12,
                            2,
                            1,
                            1337,
                            5019,
                            1,
                            1,
                            0, 
                            "UNK" }; 

SQL-query:  INSERT INTO listentries(cid,entryname,link,kind,pos,creator,visible,vers,moment,gradesystem,highscoremode,feedbackenabled,feedbackquestion) 
            VALUES ($preValuesQuiz[0], $preValuesQuiz[1], $preValuesQuiz[2], $preValuesQuiz[3], $preValuesQuiz[4], $preValuesQuiz[5], $preValuesQuiz[6], $preValuesQuiz[7], $preValuesQuiz[8], $preValuesQuiz[9], $preValuesQuiz[10],$preValuesQuiz[11],$preValuesQuiz[12]);

```

Reply:
Should we perhaps create listeentries and course at the start of the entire testing? And delete them at the end of all tests? To simplify for tester.

Reply:
i wouldn't, it will only complicate things. 
this would mean that you need to keep track on what has happened with that row during the test.
its easy enough make the inserts at the start of each test.